src/float.cr

```diff
@@ -148,8 +148,17 @@ struct Float32
     Printer.print(self, io)
   end
 
+  private def hash_normalize
+    t = to_i32
+    t == self ? t : unsafe_as(Int32)
```

Comment:
I mean more `4.25_f64.hash == 4.25_f32` ? If not, then my variant is simpler. If yes, then it is better to use more correct variant.

(note: I believe `4.1_f64 != 4.1_f32` in any case, but I could be mistaken).